src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,85 @@
+using Flux, PaddedViews
+
+basicblock(inplanes::Int, outchannels, downsample::Bool = false) = downsample ? 
+  Chain(Conv((3, 3), inplanes => outchannels[1], stride = 2),
+        BatchNorm(outchannels[1], λ = relu),
+        Conv((3, 3), outchannels[1] => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels[2], λ=relu)) : 
+  Chain(Conv((3, 3), inplanes => outchannels[1], stride = 1),
+        BatchNorm(outchannels[1], λ = relu),
+        Conv((3, 3), outchannels[1] => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels[2], λ=relu))
+
+bottleneck(inplanes::Int, outchannels, downsample::Bool = false) = downsample ?
+  Chain(Conv((1, 1), inplanes => outchannels[1], stride = 2),
+        BatchNorm(outchannels[1], λ = relu),
+        Conv((3, 3), outchannels[1] => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels[2], λ = relu),
+        Conv((1, 1), outchannels[2] => outchannels[3], stride = 1),
+        BatchNorm(outchannels[3], λ = relu)) :
+  Chain(Conv((1, 1), inplanes => outchannels[1], stride = 1),
+        BatchNorm(outchannels[1], λ = relu),
+        Conv((3, 3), outchannels[1] => outchannels[2], stride = 1, pad = 1),
+        BatchNorm(outchannels[2], λ = relu),
+        Conv((1, 1), outchannels[2] => outchannels[3], stride = 1),
+        BatchNorm(outchannels[3], λ = relu))
+
+projection(inplanes::Int, outplanes::Int, stride::Int) = Chain(Conv((1, 1), inplanes => outplanes, stride=stride),
+                                                               BatchNorm((outplanes), λ=relu))
+
+identity(inplanes::Int, outplanes::Int, stride::Int) = inplanes < outplanes ? 
+  Array -> PaddedView(0, Array, outplanes) : +
+
+function resnet(block, shortcut_config, channel_config, block_config)
+  layers = []
+  push!(layers, Conv((7, 7), 3=>inplanes, stride=(2, 2), pad=(3, 3)))
+  push!(layers, BatchNorm(inplanes, λ=relu))
+  push!(layers, MaxPool((3, 3), stride=(2, 2), pad=(1, 1)))
+  inplanes = 64
+  baseplanes = 64
+  for nrepeats in block_config
+    outplanes = baseplanes .* channel_config
+    push!(layers, block(inplanes, outplanes, true))
```

Comment:
What's the purpose of this line?

Reply:
It is incorrect. I thought of adding the first layer which won't be a skipconnection but I have already added it in the loop.